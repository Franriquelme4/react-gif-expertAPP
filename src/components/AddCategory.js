import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
        console.log('');
    }
    const handleSubmit=(e)=>{
        e.preventDefault();
        console.log(inputValue);
        if (inputValue) {
            setCategories(cat => [inputValue,...cat]);
            setInputValue('');
        }
    }

    return ( 
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                value={inputValue}
                onChange={handleInputChange}

                />
        </form>
    
    );
}
 AddCategory.propTypes = {
    setCategories:PropTypes.func.isRequired
 }
export default AddCategory;