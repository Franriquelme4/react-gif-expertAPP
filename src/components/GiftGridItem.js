import React from 'react';
import PropTypes from 'prop-types';
export const GiftGridItem = ({title,url}) => {
  return (
    <div className='card animate__animated animate__fadeIn'>
        <img src={url} alt={title}/>
        <p>{title}</p>
    </div>
  )
}

GiftGridItem.propTypes = {
  title:PropTypes.string.isRequired,
  url:PropTypes.string.isRequired
}

 /*
  installar enxyme
  1) enzyme
  2) enzyme json
  3) debe mostrar el componente correctamente
    *shallow
    *wrapper
    *wrapper . toMatchSnapshot
 
 */