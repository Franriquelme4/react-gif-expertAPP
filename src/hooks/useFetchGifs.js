import { useEffect, useState } from "react";
import { getGifts } from "../helpers/getGifts";

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });
  useEffect(()=>{
    getGifts(category)
        .then(img =>{
            setTimeout(() => {
                setState({
                    data:img,
                    loading:false
                })
            }, 2000);
           
        })
},[category])


  return state;
};
