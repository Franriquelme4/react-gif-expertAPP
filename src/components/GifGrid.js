
import React, { Fragment } from "react";
import '@testing-library/jest-dom'
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GiftGridItem } from "./GiftGridItem";
import PropTypes from 'prop-types';
export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  // se renombra

  return (
    <Fragment>
      {loading && (
        <p className="animate__animated animate__flash">Loading...</p>
      )}
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      <div className="img-container">
        {images.map((image) => (
          <GiftGridItem key={image.id} {...image} />
        ))}
      </div>
    </Fragment>
  );
};
GifGrid.propTypes={
  category:PropTypes.string.isRequired
}