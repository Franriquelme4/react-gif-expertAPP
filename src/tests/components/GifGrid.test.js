import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
//jest.mock('../../hooks/useFetchGifs');
describe('Se evalua en el componente GifGrid', () => { 
    const category = 'dog'
    let wrapper=shallow(<GifGrid category={category}/>)  
    test('Se realiza el snapchot',()=>{
    
        expect(wrapper).toMatchSnapshot();
    });


    test('debe de mostrar items  con useFetc ', () => { 
        const imgs = []
        // useFetchGifs.mockReturnValue({
        //     data: [],
        //     loanding: true,
        //   });
        //expect(wrapper).toMatchSnapshot();
          expect(wrapper.find('p').exists()).toBe(true);
            expect(wrapper.find('GiftGridItem').length).toBe(imgs.length)
     })



 })