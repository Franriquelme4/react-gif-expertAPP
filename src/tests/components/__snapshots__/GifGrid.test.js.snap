// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se evalua en el componente GifGrid Se realiza el snapchot 1`] = `
<Fragment>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <h3
    className="animate__animated animate__fadeIn"
  >
    dog
  </h3>
  <div
    className="img-container"
  />
</Fragment>
`;
